crypto/rand/randfile.c

```diff
@@ -286,8 +286,22 @@ const char *RAND_file_name(char *buf, size_t size)
         if (OPENSSL_strlcpy(buf, s, size) >= size)
             return NULL;
     } else {
+#ifdef OPENSSL_SYS_WINDOWS
+        /*
+         * We use the same env variables as GetTempFile() - but that function
+         * uses TCHARs, but getenv() gives us chars so its easier to do it this
+         * way
+         */
+        if ((s = getenv("TMP")) == NULL
+            && (s = getenv("TEMP")) == NULL
+            && (s = getenv("USERPROFILE")) == NULL
+            && (s = getenv("SYSTEMROOT")) == NULL) {
+            s = getenv("HOME");
```

Comment:
I did consider that, but my reasoning was this. %RANDFILE% is for this specific purpose, and this purpose only. Therefore that is actually handled first. %HOME% is kind of a generic thing, which although not common on Windows, might be set if you are working with Unix type tools on Windows. If it is set it probably isn't specifically for the benefit of OpenSSL - so use the preferred locations first. You could probably argue it either way.